version: '3.8'

services:
  loki-distributor:
    image: grafana/loki:2.6.1
    container_name: loki-distributor
    ports:
      # host:container
      - 3101:3100
    volumes:
      - ./config:/etc/loki/
    command: -config.file=/etc/loki/loki.yaml -target=distributor
    networks:
      - lokidemo

  loki-querier:
    image: grafana/loki:2.6.1
    container_name: loki-querier
    ports:
      # host:container
      - 3105:3100
    volumes:
      - ./config:/etc/loki/
    command: -config.file=/etc/loki/loki.yaml -target=querier
    networks:
      - lokidemo

  loki-ingester:
    image: grafana/loki:2.6.1
    container_name: loki-ingester
    ports:
      - 3100
      - "9095:9095"
    volumes:
      - ./config:/etc/loki/
    command: -config.file=/etc/loki/loki.yaml -target=ingester
    networks:
      - lokidemo

  loki-table-manager:
    image: grafana/loki:2.6.1
    container_name: loki-table-manager
    ports:
      - 3100
    volumes:
      - ./config:/etc/loki/
    command: -config.file=/etc/loki/loki.yaml -target=table-manager
    networks:
      - lokidemo

  grafana:
    image: grafana/grafana:latest
    container_name: grafana
    ports:
      - "3000:3000"
    environment:
      GF_AUTH_ANONYMOUS_ENABLED: "true"
      GF_AUTH_DISABLE_LOGIN_FORM: "true"
      GF_AUTH_ANONYMOUS_ORG_ROLE: "Admin"
    networks:
      - lokidemo
    volumes:
      - ./config/datasource.yml:/etc/grafana/provisioning/datasources/datasource.yml
      - grafana:/var/lib/grafana

volumes:
  grafana: {}

networks:
  lokidemo: 
    name: lokidemo
